service/topology_state_machine.cc

```diff
@@ -36,6 +37,36 @@ bool topology::contains(raft::server_id id) {
            left_nodes.contains(id);
 }
 
+std::set<sstring> topology::calculate_not_yet_enabled_features() const {
+    std::set<sstring> to_enable;
+    bool first = true;
+
+    auto process_nodes = [&] (const std::unordered_map<raft::server_id, replica_state>& repmap) {
+        for (const auto& rs : repmap | boost::adaptors::map_values) {
```

Comment:
I used `boost::adaptors::map_values` because capturing the structured binding like this: `[&rs]` doesn't seem to work - the compiler rejects it, not sure whether it's a clang issue or a bug in the standard. I didn't know that you could work around this in the way that you proposed - it works for me, thanks, will adjust.